tests/unit/test_finops.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import os
+
+import pytest
+
+from cost_reports import finops
+
+
+LOG = logging.getLogger(__name__)
+
+FIXTURE_DIR = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)),
+    'fixture',
+)
+
+
+@pytest.fixture()
+def parquet():
+    """Create a file-like object for the parquet fixture"""
+    fp = os.path.join(FIXTURE_DIR, 'cur.parquet')
+    rv = open(fp, 'rb')
+    if rv is None:
+        raise Exception("Error opening cur.parquet fixture")
+    return rv
+
+@pytest.fixture()
+def csv():
+    """Create a file-like object for the csv fixture"""
+    fp = os.path.join(FIXTURE_DIR, 'acc.csv')
+    rv = open(fp, 'r')
+    if rv is None:
+        raise Exception("Error opening acc.csv fixture")
+    return rv
+
+
+def test_finops(caplog, parquet, csv):
```

Comment:
I suggest factoring out the main join/groupby logic and covering with unit tests.
Also, the supported methods, 'safe_at', 'valid_value', 'cost_center_lookup', and
'group_by_cost_center' should have unit tests.  In short, all the logic in main()
should be covered by unit tests.

Reply:
done